Signal/src/ViewControllers/MessagesViewController.m

```diff
@@ -3608,6 +3684,31 @@ - (void)showErrorAlertForAttachment:(SignalAttachment * _Nullable)attachment {
                      completion:nil];
 }
 
+- (void)textViewDidChangeSize
+{
+    OWSAssert([NSThread isMainThread]);
+
+    BOOL wasAtBottom = [self isScrolledToBottom];
+    if (wasAtBottom) {
+        [self.scrollLaterTimer invalidate];
+        self.scrollLaterTimer = [NSTimer weakScheduledTimerWithTimeInterval:0.001f
+                                                                     target:self
+                                                                   selector:@selector(scrollToBottomImmediately)
+                                                                   userInfo:nil
+                                                                    repeats:NO];
+    }
+}
+
+- (void)scrollToBottomImmediately
+{
+    OWSAssert([NSThread isMainThread]);
+
+    [self.scrollLaterTimer invalidate];
+    self.scrollLaterTimer = nil;
```

Comment:
My intention is to make this "scroll alter" logic considerably simpler or unnecessary when we replace JSQMessagesView.